### PR TITLE
Telemetry improvements

### PR DESCRIPTION
**Pull request by anusapan**
1. Add an option for filtering devices
2. Changed mandatory parameters to optional paramaters.

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [x] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [x] Have **all** unit **and** integration tests passed locally? i.e. `pytest -s <project root>/azext_iot/tests`
- [x] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.

### CODE REVIEW

**Comment by digimaun** [comment]
We merged another PR...can you please pull in those changes?

**Comment by anusapan** [comment]
@digimaun I have merged the changes.

File: azext_iot/operations/digitaltwin.py [hunk 1]
```diff
@@ -175,38 +175,55 @@ def iot_digitaltwin_property_update(cmd, interface_payload, device_id,
...
+    target_interfaces = []
+    interface_id = None
+    if all([device_id, device_query]):
+        raise CLIError('You cannot use --device-id/-d and --device-query/-q at the same time!')
 
-    target_interface = next((k for k in interface_list if k['name'] == interface), None)
-    if interface and not target_interface:
-        raise CLIError('Target interface is not implemented by the device!')
+    if all([interface, device_query]):
+        raise CLIError('You cannot use --interface/-i and --device-query/-q at the same time!')
```
Comment (YingXue) [inline azext_iot/operations/digitaltwin.py]: is it a technical limitation?

File: azext_iot/operations/digitaltwin.py [hunk 2]
```diff
@@ -175,38 +175,55 @@ def iot_digitaltwin_property_update(cmd, interface_payload, device_id,
         raise CLIError(unpack_msrest_error(e))
 
 
-def iot_digitaltwin_monitor_events(cmd, device_id, interface, source_model, repo_endpoint=PNP_ENDPOINT, repo_id=None,
-                                   consumer_group='$Default', timeout=300, hub_name=None, resource_group_name=None,
-                                   yes=False, properties=None, repair=False, login=None, repo_login=None):
+def iot_digitaltwin_monitor_events(cmd, device_id=None, device_query=None, interface=None, source_model='public',
```
Comment (YingXue) [inline azext_iot/operations/digitaltwin.py]: I was wondering why source_model is not an enum type?

Comment (anusapan) [reply azext_iot/operations/digitaltwin.py]: Updated to enum

File: azext_iot/operations/digitaltwin.py (hunk 1 above)
Comment (anusapan) [reply azext_iot/operations/digitaltwin.py]: as discussed, closing this

**Pull request merged**